cmd/nerdctl/build.go

```diff
@@ -359,9 +364,30 @@ func generateBuildctlArgs(cmd *cobra.Command, buildkitHost string, platform, arg
 	if err != nil {
 		return "", nil, false, "", nil, nil, err
 	}
-	if _, err := os.Lstat(filepath.Join(absDir, file)); err != nil {
-		return "", nil, false, "", nil, nil, err
+	if file != buildkitutil.DefaultDockerfileName {
+		if _, err := os.Lstat(filepath.Join(absDir, file)); err != nil {
+			return "", nil, false, "", nil, nil, err
+		}
+	} else {
+		dockerfile, dockerfileerr := os.ReadFile(filepath.Join(absDir, file))
+		containerfile, containerfileerr := os.ReadFile(filepath.Join(absDir, Containerfile))
+		// both files exist.
+		if dockerfileerr == nil {
+			if containerfile == nil && !bytes.Equal(dockerfile, containerfile) {
+				cmd.OutOrStdout().Write([]byte("Dockerfile and Containerfile have different contents, building with Dockerfile"))
+			}
+		} else if dockerfileerr != nil {
+			if errors.Is(err, fs.ErrNotExist) && containerfileerr == nil {
+				logrus.Infof("%s does not exist, using %s as fallback", buildkitutil.DefaultDockerfileName, Containerfile)
+				file = Containerfile
+			} else {
+				return "", nil, false, "", nil, nil, err
+			}
+		} else if dockerfile != nil || containerfile != nil {
```

Comment:
We will never reach this else if

Reply:
Fixed this and added unit tests by moving to pkg as there is a lot of logic here.